character-controller.js

```diff
@@ -877,7 +879,17 @@ class UninterpolatedPlayer extends StatePlayer {
     this.actionInterpolants = {
       crouch: new BiActionInterpolant(() => this.hasAction('crouch'), 0, crouchMaxTime),
       activate: new UniActionInterpolant(() => this.hasAction('activate'), 0, activateMaxTime),
-      use: new InfiniteActionInterpolant(() => this.hasAction('use'), 0),
+      use: new InfiniteActionInterpolant(() => {
+        const useAction = this.getAction('use')
+        if (useAction) {
+          if (useAction.needResetUseTime) {

```

Comment:
I don't think it's legal to change the action.

This will not be picked up when the action is replicated over network.

Reply:
If there is some state that needs to be tracked with the action then I think that state needs to live with whatever class is handling the action.

Reply:
Add actions `needEndUse`, `needContinueCombo` and `needResetUseTime`, instead of modify `useAction`. https://github.com/webaverse/app/pull/2804/commits/c0838b61112912eb0bb2f448c71c626d33917a7f
Is this proper?

Reply:
No I don't think so. Actions are for actions the player is doing. These are not actions.

Reply:
@avaer Yeah, they are just states.
How about make them as player's state variables? https://github.com/webaverse/app/pull/2804/commits/0a325db0917c3c6a0876f0c3aef482fb62e2f6fb